cmd/profiles.go

```diff
@@ -0,0 +1,302 @@
+	Pretty     bool
+}
+
+// ProfilesCmd handles profile operations independent of cobra.
+type ProfilesCmd struct {
+	profiles ProfilesService
+}
+
+func (p ProfilesCmd) List(ctx context.Context) error {
+	pterm.Info.Println("Fetching profiles...")
+	items, err := p.profiles.List(ctx)
+	if err != nil {
+		return util.CleanedUpSdkError{Err: err}
+	}
+	if items == nil || len(*items) == 0 {
+		pterm.Info.Println("No profiles found")
+		return nil
+	}
+	rows := pterm.TableData{{"Profile ID", "Name", "Created At", "Updated At", "Last Used At"}}
+	for _, prof := range *items {
+		name := prof.Name
+		if name == "" {
+			name = "-"
+		}
+		rows = append(rows, []string{
+			prof.ID,
+			name,
+			util.FormatLocal(prof.CreatedAt),
+			util.FormatLocal(prof.UpdatedAt),
+			util.FormatLocal(prof.LastUsedAt),
+		})
+	}
+	printTableNoPad(rows, true)
+	return nil
+}
+
+func (p ProfilesCmd) Get(ctx context.Context, in ProfilesGetInput) error {
+	item, err := p.profiles.Get(ctx, in.Identifier)
+	if err != nil {
+		return util.CleanedUpSdkError{Err: err}
+	}
+	if item == nil || item.ID == "" {
+		pterm.Error.Printf("Profile '%s' not found\n", in.Identifier)
+		return nil
+	}
+	name := item.Name
+	if name == "" {
+		name = "-"
+	}
+	rows := pterm.TableData{{"Property", "Value"}}
+	rows = append(rows, []string{"ID", item.ID})
+	rows = append(rows, []string{"Name", name})
+	rows = append(rows, []string{"Created At", util.FormatLocal(item.CreatedAt)})
+	rows = append(rows, []string{"Updated At", util.FormatLocal(item.UpdatedAt)})
+	rows = append(rows, []string{"Last Used At", util.FormatLocal(item.LastUsedAt)})
+	printTableNoPad(rows, true)
+	return nil
+}
+
+func (p ProfilesCmd) Create(ctx context.Context, in ProfilesCreateInput) error {
+	params := kernel.ProfileNewParams{}
+	if in.Name != "" {
+		params.Name = kernel.Opt(in.Name)
+	}
+	item, err := p.profiles.New(ctx, params)
+	if err != nil {
+		return util.CleanedUpSdkError{Err: err}
+	}
+	name := item.Name
+	if name == "" {
+		name = "-"
+	}
+	rows := pterm.TableData{{"Property", "Value"}}
+	rows = append(rows, []string{"ID", item.ID})
+	rows = append(rows, []string{"Name", name})
+	rows = append(rows, []string{"Created At", util.FormatLocal(item.CreatedAt)})
+	rows = append(rows, []string{"Last Used At", util.FormatLocal(item.LastUsedAt)})
+	printTableNoPad(rows, true)
+	return nil
+}
+
+func (p ProfilesCmd) Delete(ctx context.Context, in ProfilesDeleteInput) error {
+	if !in.SkipConfirm {
+		// Try to resolve for a nicer message; avoid prompting for missing entries
+		list, err := p.profiles.List(ctx)
+		if err != nil {
+			return util.CleanedUpSdkError{Err: err}
+		}
+		var found *kernel.Profile
+		if list != nil {
+			for _, pr := range *list {
+				if pr.ID == in.Identifier || (pr.Name != "" && pr.Name == in.Identifier) {
+					cp := pr
+					found = &cp
+					break
+				}
+			}
+		}
+		if found == nil {
+			pterm.Error.Printf("Profile '%s' not found\n", in.Identifier)
+			return nil
+		}
```

Comment:
`List` over `Get` is a little surprising here tbh. This probably is negligible at first but would likely be worse in perf with deeper adoption